datadog/agent/install.go

```diff
@@ -142,3 +144,20 @@ func installIntegrations(
 
 	return lastCommand, utils.StrHash(parts...), nil
 }
+
+type ClientData struct {
+	Connection utils.Connection
+}
+
+func (installer *Installer) GetClientDataDeserializer() func(auto.UpResult) (*ClientData, error) {
+	vmDataDeserializer := installer.vm.GetClientDataDeserializer()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why not returning `installer.vm.GetClientDataDeserializer` ?

Reply:
The goal is to have different types between `Installer` and `installer.vm.GetClientDataDeserializer` to enforce strong typing.